MediaAdmin/EventSubscriber/MediaCreatedSubscriber.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace OpenOrchestra\MediaAdmin\EventSubscriber;
+
+use Doctrine\Common\Persistence\ObjectManager;
+use OpenOrchestra\MediaAdmin\FileAlternatives\FileAlternativesManager;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use OpenOrchestra\MediaAdmin\MediaEvents;
+use OpenOrchestra\MediaAdmin\Event\MediaEvent;
+
+/**
+ * Class MediaCreatedSubscriber
+ */
+class MediaCreatedSubscriber implements EventSubscriberInterface
+{
+    public $medias = array();
+    protected $fileAlternativesManager;
+    protected $objectManager;
+
+    /**
+     * @param fileAlternativesManager $fileAlternativesManager
+     * @param ObjectManager           $objectManager
+     */
+    public function __construct(FileAlternativesManager $fileAlternativesManager, ObjectManager $objectManager)
+    {
+        $this->fileAlternativesManager = $fileAlternativesManager;
+        $this->objectManager = $objectManager;
+    }
+
+    /**
+     * @param MediaEvent $event
+     */
+    public function addMedia(MediaEvent $event)
+    {
+        $media = $event->getMedia();
+        $this->medias[] = $media;
+    }
+
+    /**
+     * Generate images
+     */
+    public function generateAlternatives()
+    {
+        /** @var MediaInterface $media */
+        foreach ($this->medias as $media) {
+            $media = $this->fileAlternativesManager->generateThumbnail($media);
```

Comment:
As a matter of fact, the object passed is always a reference, and if you modify it in the manager then do the flush, all the modification will be send to the database